test cases/common/183 as-needed/libB.cpp

```diff
@@ -0,0 +1,13 @@
+#define BUILDING_DLL
```

Comment:
You need to define `BUILDING_DLL` after you include `libA.h`. Since `libA` is external towards `libB`, `dllimport` has to be used instead.
Also there is nothing exported from `libB` so once you fix issue above you will face missing `libB.lib` file most likely. You could add `bool DLL_PUBLIC lib_b_dummy() { return true; }` for example to mitigate this.